pyro/nn/auto_reg_nn.py

```diff
@@ -104,39 +102,66 @@ class AutoRegressiveNN(nn.Module):
     """
     An implementation of a MADE-like auto-regressive neural network.
 
-    Reference:
-    MADE: Masked Autoencoder for Distribution Estimation [arXiv:1502.03509]
-    Mathieu Germain, Karol Gregor, Iain Murray, Hugo Larochelle
+    Example usage:
+    >>> x = torch.randn(100, 10)
+    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1])
+    >>> p = arn(x)  # 1 parameters of size (100, 10)
+    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1, 1])
+    >>> m, s = arn(x) # 2 parameters of size (100, 10)
+    >>> arn = AutoRegressiveNN(10, [50], param_dims=[1, 5, 3])
+    >>> a, b, c = arn(x) # 3 parameters of sizes, (100, 1, 10), (100, 5, 10), (100, 3, 10)
 
     :param input_dim: the dimensionality of the input
     :type input_dim: int
-    :param hidden_dim: the dimensionality of the hidden units
-    :type hidden_dim: int
-    :param output_dim_multiplier: the dimensionality of the output is given by input_dim x output_dim_multiplier.
-        specifically the shape of the output for a single vector input is [output_dim_multiplier, input_dim].
-        for any i, j in range(0, output_dim_multiplier) the subset of outputs [i, :] has identical
-        autoregressive structure to [j, :]. defaults to `1`
-    :type output_dim_multiplier: int
+    :param hidden_dims: the dimensionality of the hidden units per layer
+    :type hidden_dims: list[int]
+    :param param_dims: shape the output into parameters of dimension (p_n, input_dim) for p_n in param_dims
+        when p_n > 1 and dimension (input_dim) when p_n == 1
+    :type param_dims: list[int]
     :param permutation: an optional permutation that is applied to the inputs and controls the order of the
         autoregressive factorization. in particular for the identity permutation the autoregressive structure
         is such that the Jacobian is upper triangular. By default this is chosen at random.
     :type permutation: torch.LongTensor
+    :param skip_connections: Whether to add skip connections from the input to the output.
+    :type skip_connections: bool
+    :param nonlinearity: The nonlinearity to use in the feedforward network.
+    :type nonlinearity: torch.nn.module
+
+    Reference:
+
+    MADE: Masked Autoencoder for Distribution Estimation [arXiv:1502.03509]
+    Mathieu Germain, Karol Gregor, Iain Murray, Hugo Larochelle
+
     """
 
-    def __init__(self, input_dim, hidden_dim, output_dim_multiplier=1, permutation=None, skip_connections=False,
-                 num_layers=1, nonlinearity=nn.ReLU()):
+    def __init__(
+            self,
+            input_dim,
+            hidden_dims,
+            param_dims=[1, 1],
```

Comment:
this default `param_dims=[1, 1]` should be specified in the doc string

Reply:
Done!